File: C#/Day1/Day1.cs
using System.IO;

namespace AdventOfCode2020.Day1
{
    public class Part1
    {
        public static int SolvePuzzle(string[] input)
        {
                int[] numbers = Array.ConvertAll(input, int.Parse);

            for (int i = 0; i < numbers.Length; i++)
            {
                for (int j = i + 1; j < numbers.Length; j++)
                {
                    if (numbers[i] + numbers[j] == 2020)
                    {
                        return numbers[i] * numbers[j];
                    }
                }
            }

           return -1;
        }
    }
    public class Part2
    {
        public static int SolvePuzzle(string[] input)
        {
                int[] numbers = Array.ConvertAll(input, int.Parse);

            for (int i = 0; i < numbers.Length; i++)
            {
                for (int j = i + 1; j < numbers.Length; j++)
                {
                     for (int k = 0; k < numbers.Length; k++)
            {
                    if (numbers[i] + numbers[j] + numbers[k] == 2020)
                    {
                        return numbers[i] * numbers[j] * numbers[k];
                    }
                }
            }
            }
           return -1;
        }
    }
}

File: Clojure/2018/aoc-2018/src/aoc_2018/day1.clj
(ns aoc-2018.day1
  (:require
   [aoc-2018.day1 :as day1]
   [aoc-2018.core :as core]))

(defn part1 []
  (let [numbers (core/parse-input "day1")
        total-sum (reduce + numbers)]
    total-sum))

(defn part2 []
  (loop [seen #{}
         current-sum 0
         remaining (cycle (core/parse-input "day1"))]
    (let [next-number (first remaining)
          new-sum (+ current-sum next-number)]
      (if (seen new-sum)
        new-sum
        (recur (conj seen new-sum) new-sum (rest remaining))))))

(defn solve []
  (println "Day 1: Part 1:")
  (println (day1/part1))

  (println "Day 1: Part 2:")
  (println (day1/part2)))
